src/parsers/manifest/dash/attach_trickmode_track.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IAdaptationType } from "../../../manifest";
+import {
+  IParsedAdaptation,
+  IParsedAdaptations,
+} from "../types";
+
+const KNOWN_ADAPTATION_TYPES : IAdaptationType[] = ["audio", "video", "text", "image"];
+
+/**
+ * Attach trick mode tracks to adaptations.
+ * @param {Object} adaptations
+ * @param {Array.<Object>} trickModeTracks
+ * @returns {void}
+ */
+function attachTrickModeTrack(
+  adaptations: IParsedAdaptations,
+  trickModeTracks: Array<{ adaptation: IParsedAdaptation;
+                           trickModeAttachedAdaptationIds: string[]; }>
+): void {
+  for (let i = 0; i < trickModeTracks.length; i++) {
+    const { adaptation, trickModeAttachedAdaptationIds } = trickModeTracks[i];
+    for (let m = 0; m < trickModeAttachedAdaptationIds.length; m++) {
+      const trickModeAttachedAdaptationId = trickModeAttachedAdaptationIds[m];
+      for (let j = 0; j < KNOWN_ADAPTATION_TYPES.length; j++) {
+        const adaptationType = KNOWN_ADAPTATION_TYPES[j];
+        const adaptationsByType = adaptations[adaptationType];
+        if (adaptationsByType !== undefined) {
+          for (let k = 0; k < adaptationsByType.length; k++) {
+            const adaptationByType = adaptationsByType[k];
+            if (adaptationByType.id === trickModeAttachedAdaptationId) {
+              if (adaptationByType.trickModeTracks === undefined) {
+                adaptationByType.trickModeTracks = [];
+              }
+              adaptationByType.trickModeTracks?.push(adaptation);
```

Comment:
The `?` shouldn't be needed. We should not risk this function doing nothing

Reply:
👍